gsa/src/web/components/powerfilter/filterdialog.js

```diff
@@ -82,22 +83,23 @@ class FilterDialog extends React.Component {
   }
 
   handleSave() {
-    let {filter, filterName = '', filterstring, saveNamedFilter} = this.state;
+    const {filter, filterName = '', filterstring, saveNamedFilter} = this.state;
     const {onFilterChanged, onCloseClick} = this.props;
 
-    filter = Filter.fromString(filterstring, filter);
+    const newFilter = Filter.fromString(filterstring);
+    newFilter._merge(filter);
```

Comment:
Would be nice if we can rename the Filter model `_merge` method into `merge` too. Could be done in another PR.

Reply:
I've used the python scheme for method names where an underscore marks internal methods, which shouldn't be used from the outside of the module. This isn't valid for _merge anymore :-)

Reply:
Cool. I'll make another PR for that.